docs/evaluate/development-features/application-message-passing.mdx

```diff
@@ -0,0 +1,44 @@
+---
+id: application-message-passing
+title: Application Message Passing - Temporal development feature
+description: Enhance your Workflows with Signals and Queries, allowing dynamic responses to external events and real-time state access for comprehensive monitoring and tracking.
+sidebar_label: Application Message Passing
+tags:
+  - signals
+  - queries
+  - workflows
+  - activities
+  - workers
+keywords:
+  - temporal application
+  - workflow design
+  - business logic activities
+  - temporal workers
+  - temporal SDK tutorial
+  - go sdk temporal guide
+  - java sdk temporal guide
+  - python sdk temporal guide
+  - typescript sdk temporal guide
+  - learning temporal workflows
+  - developing with temporal
+  - temporal workflow execution
+  - temporal activity management
+  - worker process execution
+  - signals
+  - queries
+---
+
+Integrating Signals and Queries is an essential skill to build responsive applications. Signals allow Workflows to react to events at runtime, while Queries enable on-demand data retrieval from ongoing Workflows without disrupting their execution.
+
+**Signals**: Signals allow you to send messages asynchronously to running Workflow Executions, enabling you to change their state and control their flow in real-time. This feature is particularly useful in scenarios where Workflows must react to external events or user interactions.
+
+**Queries**: Queries provide a way to access and monitor the internal state of your Workflow Executions in real-time. This is essential for monitoring the progress of long-running Workflows and retrieving their results.
+
+See the SDK feature guides for implementation details:
+
+- [Go SDK Messaging Feature Guide](/dev-guide/go/messages)
+- [Java SDK Messaging Feature Guide](/dev-guide/java/messages)
+- [Python SDK Messaging Feature Guide](/dev-guide/python/messages)
+- [TypeScript SDK Messaging Feature Guide](/dev-guide/typescript/messages)
```

Comment:
Vercel is blocked right now since these pages are not merged in yet. We will merge this PR after these pages are merged.